File: python/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s_len: int = len(s)
        t_len: int = len(t)

        if s_len != t_len:
            return False

        s_dict = {}
        t_dict = {}

        for i in range(s_len):
            s_dict[s[i]] = s_dict.get(s[i], 0) + 1
            t_dict[t[i]] = t_dict.get(t[i], 0) + 1

        return s_dict == t_dict

File: cpp/validAnagram.cpp
#include <string>
#include <unordered_map>

using namespace std;

class Solution {
public:
  bool isAnagram(string s, string t) {
    if (s.length() != t.length()) {
      return false;
    }

    unordered_map<char, int> map;

    for (int i = 0; i < s.length(); i++) {
      map[s.at(i)]++;
    }

    for (int j = 0; j < t.length(); j++) {
      map[t.at(j)]--;
    }

    for (auto entry : map) {
      if (entry.second != 0) {
        return false;
      }
    }

    return true;
  }
};
